config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
+	ret, err := convertVal(value, targetType)
+	if ret == nil {
+		return nil, fmt.Errorf("can't convert %v to %v", reflect.TypeOf(value), targetType)
+	}
+	return ret, err
+}
+
+func convertStructValue(value interface{}, targetType reflect.Type, fieldType reflect.Type, fieldValue reflect.Value) (interface{}, error) {
+	if ret, err := convertVal(value, targetType); ret != nil {
+		return ret, err
+	}
+
+	// The fieldType is probably custom type here. We will try and set the fieldValue by
+	// the type of custom type
+	kind := fieldType.Kind()
+	switch kind {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
```

Comment:
Not sure what you mean. I want to avoid having hardcoded lists of int types and float types everywhere.
So we would have:
kind := fieldType.Kind()
 	switch {
 	case isInt(kind):
               fieldValue.SetInt(int64(value.(int)))
        case isUInt(kind):
		fieldValue.SetUint(uint64(value.(int)))
        etc.